radvdump.c

```diff
@@ -244,6 +244,15 @@ static void print_ff(unsigned char *msg, int len, struct sockaddr_in6 *addr, int
 				printf("\tHomeAgentLifetime %hu;\n", ntohs(ha_info->lifetime));
 			break;
 		}
+		case ND_OPT_CAPTIVE_PORTAL: {
+			char *opt_captive_portal = (char *)opt_str+2;
+			char *captive_portal = strndup(opt_captive_portal, optlen-2);
+
+			printf("\tAdvCaptivePortalAPI \"%s\";\n", captive_portal);
+
+			free(captive_portal);
```

Comment:
I don't understand this `free()`.
Probably due me being more sysadmin than C programmer.

Reply:
The strndup() just above the directly shown lines allocates memory, that needs to be freed.
I used the strndup() call as this code part if not performance critical, and the function provides easy null-terminating of the string.